pkg/virt-handler/vm.go

```diff
@@ -1272,29 +1272,26 @@ func (d *VirtualMachineController) updateFSFreezeStatus(vmi *v1.VirtualMachineIn
 
 }
 
-func IsoGuestVolumePath(vmi *v1.VirtualMachineInstance, volume *v1.Volume) (string, bool) {
-	var volPath string
-
-	basepath := "/var/run"
-	if volume.CloudInitNoCloud != nil {
-		volPath = filepath.Join(basepath, "kubevirt-ephemeral-disks", "cloud-init-data", vmi.Namespace, vmi.Name, "noCloud.iso")
-	} else if volume.CloudInitConfigDrive != nil {
-		volPath = filepath.Join(basepath, "kubevirt-ephemeral-disks", "cloud-init-data", vmi.Namespace, vmi.Name, "configdrive.iso")
-	} else if volume.ConfigMap != nil {
-		volPath = config.GetConfigMapDiskPath(volume.Name)
-	} else if volume.DownwardAPI != nil {
-		volPath = config.GetDownwardAPIDiskPath(volume.Name)
-	} else if volume.Secret != nil {
-		volPath = config.GetSecretDiskPath(volume.Name)
-	} else if volume.ServiceAccount != nil {
-		volPath = config.GetServiceAccountDiskPath()
-	} else if volume.Sysprep != nil {
-		volPath = config.GetSysprepDiskPath(volume.Name)
-	} else {
+func IsoGuestVolumePath(namespace, name string, volume *v1.Volume) (string, bool) {
+	const basepath = "/var/run"
+	switch {
+	case volume.CloudInitNoCloud != nil:
+		return filepath.Join(basepath, "kubevirt-ephemeral-disks", "cloud-init-data", namespace, name, "noCloud.iso"), true
+	case volume.CloudInitConfigDrive != nil:
+		return filepath.Join(basepath, "kubevirt-ephemeral-disks", "cloud-init-data", namespace, name, "configdrive.iso"), true
+	case volume.ConfigMap != nil:
+		return config.GetConfigMapDiskPath(volume.Name), true
+	case volume.DownwardAPI != nil:
+		return config.GetDownwardAPIDiskPath(volume.Name), true
+	case volume.Secret != nil:
+		return config.GetSecretDiskPath(volume.Name), true
+	case volume.ServiceAccount != nil:
+		return config.GetServiceAccountDiskPath(), true
+	case volume.Sysprep != nil:
+		return config.GetSysprepDiskPath(volume.Name), true
+	default:
 		return "", false
```

Comment:
nit: we can eliminate the bool and simply return an empty string when it wasn't found

Reply:
Done